collector-http_test.go

```diff
@@ -0,0 +1,123 @@
+package zipkintracer
+
+import (
+	"testing"
+	"time"
+
+	"io/ioutil"
+	"net/http"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/openzipkin/zipkin-go-opentracing/_thrift/gen-go/zipkincore"
+	"sync"
+)
+
+func TestHttpCollector(t *testing.T) {
+	server := newHTTPServer(t)
+	c, err := NewHTTPCollector("http://localhost:10000/zipkin")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		serviceName  = "service"
+		methodName   = "method"
+		traceID      = int64(123)
+		spanID       = int64(456)
+		parentSpanID = int64(0)
+		value        = "foo"
+	)
+
+	span := makeNewSpan("1.2.3.4:1234", serviceName, methodName, traceID, spanID, parentSpanID, true)
+	annotate(span, time.Now(), "foo", nil)
+	if err := c.Collect(span); err != nil {
+		t.Errorf("error during collection: %v", err)
+	}
+
+	// Need to yield to the select loop to accept the send request, and then
+	// yield again to the send operation to write to the socket. I think the
+	// best way to do that is just give it some time.
+
+	deadline := time.Now().Add(1 * time.Second)
+	for {
+		if time.Now().After(deadline) {
+			t.Fatalf("never received a span")
+		}
+		if want, have := 1, len(server.spans()); want != have {
+			time.Sleep(time.Millisecond)
+			continue
+		}
+		break
+	}
+
+	gotSpan := server.spans()[0]
+	if want, have := methodName, gotSpan.GetName(); want != have {
+		t.Errorf("want %q, have %q", want, have)
+	}
+	if want, have := traceID, gotSpan.TraceID; want != have {
+		t.Errorf("want %d, have %d", want, have)
+	}
+	if want, have := spanID, gotSpan.ID; want != have {
+		t.Errorf("want %d, have %d", want, have)
+	}
+	if want, have := parentSpanID, *gotSpan.ParentID; want != have {
+		t.Errorf("want %d, have %d", want, have)
+	}
+
+	if want, have := 1, len(gotSpan.GetAnnotations()); want != have {
+		t.Fatalf("want %d, have %d", want, have)
+	}
+
+	gotAnnotation := gotSpan.GetAnnotations()[0]
+	if want, have := value, gotAnnotation.GetValue(); want != have {
+		t.Errorf("want %q, have %q", want, have)
+	}
+}
+
+type httpServer struct {
+	t           *testing.T
+	zipkinSpans []*zipkincore.Span
+	mutex       sync.RWMutex
+}
+
+func (s *httpServer) spans() []*zipkincore.Span {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+	return s.zipkinSpans
+}
+
+func newHTTPServer(t *testing.T) *httpServer {
+	server := &httpServer{
+		t:           t,
+		zipkinSpans: make([]*zipkincore.Span, 0),
+		mutex:       sync.RWMutex{},
+	}
+	http.HandleFunc("/zipkin", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
/api/v1/spans

also, validate the content-type header, as it is important (as without the content type, zipkin assumes it is json)